dali/pipeline/operators/operator.h

```diff
@@ -212,6 +232,101 @@ class Operator : public OperatorBase {
   template <typename B = Backend>
   typename std::enable_if<!std::is_same<B, GPUBackend>::value>::type
   SyncHelper(int /*unused*/, Workspace<B> */*unused*/) {}
+
+  template <typename B = Backend>
+  typename std::enable_if<std::is_same<B, GPUBackend>::value>::type
+  Flatten(Workspace<Backend> *ws) {
+    seq_sizes_.resize(input_sets_, 1);
+    for (int i = 0; i < input_sets_; ++i) {
+      auto &input = ws->template MutableInput<Backend>(i);
+
+      const std::vector<Dims> old_shapes = input.shape();
+      // only if Tensors are 4-d
+      if (old_shapes[0].size() == 4) {
```

Comment:
Why this assumption? What about grayscale videos? It'd be better to check layout.

Reply:
Was planning to change it, for testing